### PR TITLE
Resolved an issue where path is not encoded.

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/src/datalake_directory_client.cpp [hunk 1]
```diff
@@ -125,18 +125,20 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
     auto builder = m_dfsUri;
-    builder.AppendPath(path, true);
+    builder.AppendPath(path);
     auto blobClient = m_blobClient;
-    blobClient.m_blobUrl.AppendPath(path, true);
-    return FileClient(std::move(builder), std::move(blobClient), m_pipeline);
+    blobClient.m_blobUrl.AppendPath(path);
+    auto blockBlobClient = blobClient.GetBlockBlobClient();
+    return FileClient(
+        std::move(builder), std::move(blobClient), std::move(blockBlobClient), m_pipeline);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_directory_client.cpp:133]: For the record, I don't think it's right for `FileClient` to include two blob clients. But I can live with it as long as it works.

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp [hunk 2]
```diff
@@ -151,22 +151,25 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
+    auto blockBlobClient = blobClient.GetBlockBlobClient();
+    return FileClient(
+        std::move(builder), std::move(blobClient), std::move(blockBlobClient), m_pipeline);
   }
 
   DirectoryClient FileSystemClient::GetDirectoryClient(const std::string& path) const
   {
     auto builder = m_dfsUri;
-    builder.AppendPath(path, true);
+    builder.AppendPath(path);
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp:172]: @Jinming-Hu, similar to https://github.com/Azure/azure-sdk-for-cpp/pull/655#discussion_r489134444, are you planning to add tests to cover these as well, as part of https://github.com/Azure/azure-sdk-for-cpp/issues/664?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp:172]: @katmsft is responsible for datalake service.

Comment (ahsonkhan) [reply sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp:172]: @katmsft - add unit tests?

**Pull request merged**